docs/book/how-to/popular-integrations/azure-guide.md

```diff
@@ -0,0 +1,196 @@
+---
+description: A simple guide to create an Azure stack to run your ZenML pipelines
+---
+
+# Run on Azure
+
+This page aims to quickly set up a minimal production stack on Azure. With 
+just a few simple steps, you will set up a resource group, a service principal 
+with correct permissions and the relevant ZenML stack and components.
+
+{% hint style="info" %}
+Would you like to skip ahead and deploy a full Azure ZenML cloud stack already?
+
+Check out the
+[in-browser stack deployment wizard](../../how-to/stack-deployment/deploy-a-cloud-stack.md),
+the [stack registration wizard](../../how-to/stack-deployment/register-a-cloud-stack.md),
+or [the ZenML Azure Terraform module](../../how-to/stack-deployment/deploy-a-cloud-stack-with-terraform.md)
+for a shortcut on how to deploy & register this stack.
+{% endhint %}
+
+To follow this guide, you need:
+
+* An active Azure account.
+* ZenML [installed](../../getting-started/installation.md).
+* ZenML `azure` integration installed with `zenml integration install azure`.
+
+## 1. Set up proper credentials
+
+You can start by [creating a service principal by creating an app registration](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/manage-azureml-service/authentication-in-azureml/authentication-in-azureml.ipynb) 
+on Azure:
+
+1. Go to the App Registrations on the Azure portal.
+2. Click on `+ New registration`,
+3. Give it a name and click register.
+
+![Azure App Registrations](../../.gitbook/assets/azure_1.png)
+
+![Azure App Registrations](../../.gitbook/assets/azure_2.png)
+
+Once you create the service principal, you will get an Application ID and 
+Tenant ID as they will be needed later.
+
+Next, go to your service principal and click on the `Certificates & secrets` in
+the `Manage` menu. Here, you have to create a client secret. Note down the 
+secret value as it will be needed later.
+
+![Azure App Registrations](../../.gitbook/assets/azure_3.png)
+
+## 2. Create a service connector
+
+Once you have created the service principal and the client secret, you can 
+go ahead and create [a ZenML Azure Service Connector](../../how-to/auth-management/azure-service-connector.md).
+
+```bash
+zenml service-connector register azure_connector --type azure \
+  --auth-method service-principal \
+  --client_secret=<CLIENT_SECRET> \
+  --tenant_id=<TENANT_ID> \
+  --client_id=<APPLICATION_ID>
+```
+
+You will use this service connector later on to connect your components with 
+proper authentication.
+
+## 3. Create a resource group and the AzureML instance
```

Comment:
Is there a CLI-driven way to do this? Generally it gets updated less often than the UI, so less updates needed for the docs. Also easier for someone to navigate, particular a core user of ZenML. i.e. using the Azure CLI.

Reply:
I am sure there must be a way. But so far, in my experiences with Azure, the tutorials were also very UI heavy and I find it to be a bit more comfortable for the user. So, if it is ok I would leave it like this.

Reply:
For all our other tutorials we switched to CLI-driven. That was for a reason, if I'm not mistaken. I guess I'm fine if you want to get it out quickly, but I'd suggest that you create a ticket to use the CLI for the Azure resource creation so that we don't let it slip off the radar / we do it at some point.